source/type/type.h

```diff
@@ -980,18 +984,63 @@ struct Calendar : public Nameable, public Header {
  *
  */
 struct MetaVehicleJourney: public Header, HasMessages {
-    //TODO if needed use a flat_enum_map
-    std::vector<VehicleJourney*> base_vj;
-    std::vector<VehicleJourney*> adapted_vj;
-    std::vector<VehicleJourney*> real_time_vj;
+
 
     /// map of the calendars that nearly match union of the validity pattern
     /// of the theoric vj, key is the calendar name
     std::map<std::string, AssociatedCalendar*> associated_calendars;
 
     template<class Archive> void serialize(Archive & ar, const unsigned int ) {
-        ar & idx & uri & base_vj & adapted_vj & real_time_vj & associated_calendars & impacts;
+        ar & idx & uri;
+        for (const auto& rt_vjs: rtlevel_to_vjs_map) {
+            ar & rt_vjs.second;
+        }
+        ar & associated_calendars & impacts;
+    }
+
+    // TODO XL: this function should be called in places where add_vj is called, because MetaVehicleJourney will
+    //          have the ownership of vjs instead of data
+    VehicleJourney* create_vj(/*args,*/ RTLevel level);
+
+    void add_vj(VehicleJourney* vj, RTLevel level) {
+        rtlevel_to_vjs_map[level].push_back(vj);
     }
+
+
+    VehicleJourney* get_first_vj_at(navitia::type::RTLevel level) const{
+        assert(!rtlevel_to_vjs_map[level].empty());
+        return rtlevel_to_vjs_map[level].front();
+    }
+
+    template<typename T>
+    void for_all_vjs(T fun) const{
+        for (const auto& rt_vjs: rtlevel_to_vjs_map) {
+            auto& vjs = rt_vjs.second;
+            boost::for_each(vjs, [&](VehicleJourney* vj){fun(*vj);});
+        }
+    }
+
+    template<typename T>
+    void for_vjs_at_rt_level(RTLevel level, T fun ) const{
```

Comment:
Yes Maybe we can use the code in get_vj_at_date to do iether an iterator on vj* from a RTLevel or a functor.